File: web/plugins/binary_to_number.js
import { WebPlugin } from "../engine/plugin.js";

export class BinaryToNumberPlugin extends WebPlugin {
    constructor() {
        super("2進数を数字にする");
    }

    async execute(line, context) {
        const match = line.match(/^2進数を数字にする\s+「(.+)」$/);
        if (!match) return;

        context.runtime.write(
            parseInt(match[1], 2)
        );
    }
}

File: pc/plugins/binary_to_number.py
import re
from core.plugin import Plugin


class BinaryToNumberPlugin(Plugin):
    def translate(self, line):
        match = re.fullmatch(r'2進数を数字にする\s+「(.+)」', line)
        if not match:
            return None

        value = match.group(1)
        return f"print(int({value!r}, 2))"